tests/js/t20235.nim

```diff
@@ -0,0 +1,11 @@
+discard """
+  action: "run"
+  output: "0 4"
+"""
+
+proc main =
+  var s = ""
+  s.setLen(4)
```

Comment:
what if it's `s.setLen procThatEchosAndReturns(3)`?

Reply:
As in something like this?
```nim
import sequtils

proc procThatEchosAndReturns(x: int): int =
  echo "Test"
  x

var s = ""
s.setLen procThatEchosAndReturns(3)
echo s.mapIt(it.ord)
```
Outputs
```
Test
@[0, 0, 0]
```

Would this somehow cause problems?

Reply:
The parameter is assigned into a temporary so the parameter is not called multiple times

Simplified version of JS generation
```js
if (s[0].length < (Temporary1 = chckRange(procThatEchosAndReturns(s[0], 3), 0, 2147483647), Temporary1)) { for (var i = s[0].length; i < Temporary1; ++i) s[0].push(0); }
else {s[0].length = Temporary1; };
```

Reply:
Ok.